backend/src/graphql/resolver/TransactionResolver.test.ts

```diff
@@ -0,0 +1,318 @@
+          errors: [new GraphQLError('The recipient account is not activated')],
+        }),
+      )
+    })
+  })
+
+  describe('errors in the transaction itself', () => {
+    beforeAll(async () => {
+      await mutate({
+        query: login,
+        variables: bobData,
+      })
+    })
+
+    describe('sender and recipient are the same', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'bob@baumeister.de',
+              amount: 100,
+              memo: 'test',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('Sender and Recipient are the same.')],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith('Sender and Recipient are the same.')
+      })
+    })
+
+    describe('memo text is too long', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: 100,
+              memo: 'test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test t',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('memo text is too long (255 characters maximum)')],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith('memo text is too long: memo.length=256 > 255')
+      })
+    })
+
+    describe('memo text is too short', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: 100,
+              memo: 'test',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('memo text is too short (5 characters minimum)')],
+          }),
+        )
+      })
+
+      it('logs the error thrown', () => {
+        expect(logger.error).toBeCalledWith('memo text is too short: memo.length=4 < 5')
+      })
+    })
+
+    describe('user has not enough GDD', () => {
+      it('throws an error', async () => {
+        expect(
+          await mutate({
+            mutation: sendCoins,
+            variables: {
+              email: 'peter@lustig.de',
+              amount: 100,
+              memo: 'testing',
+            },
+          }),
+        ).toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError(`user hasn't enough GDD or amount is < 0`)],
```

Comment:
What happens if we send an negative amount?. Can we have a test for this

Reply:
validate.ts file is very messy for me, and clearly not working, cause a negative amount can be sent, you were right

Reply:
solved now, still a mess that function... from my point of view it should be reviewed